### ISSUE TITLE
Autobinding of named methods that shadow globals fails

### ISSUE THREAD

**Issue by kevinb7** [facebook/react#3928]
The bound method inside the guard method gets bound to ```window``` for some reason.

This fails:
```
var TestComponent = React.createClass({
    intervals: [],
    render: function() {
        return React.createElement("div");
    },
    setInterval: function setInterval(fn, ms) {
        this.intervals.push(setInterval(fn, ms));
    },
    componentDidMount: function() {
        this.setInterval(function () {
            console.log("hello");
        }, 1000);
    }
});

var testComponent = React.createElement(TestComponent, null);

React.render(testComponent, document.body);
```
with the error message:
```Uncaught TypeError: Cannot read property 'push' of undefined (index.js:7)```

I set a break point in ```setInterval``` and the first time it's called ```this``` is set to an instance of a react class.  It hit play and the method gets hit a second time and ```this``` is set to ```window```.

Changing ```setInterval: function setInterval(fn, ms) { ... }``` to ```setInterval: function (fn, ms) { ... }``` fixes the issue.  This is code that's being generated by Babel so removing named functions is a little more work than just removing them from the source code.  It would be nice if React supported this syntax, but I don't know how much work is involved.

**Comment by spicyj** [comment]
This is because `setInterval` now refers to the function itself, instead of the global setInterval. Filed https://github.com/babel/babel/issues/1598 upstream.

**Comment by spicyj** [comment]
(Autobinding still works, but then this.setInterval calls setInterval (itself) again without context and you get the error.)

**Comment by sebmck** [comment]
Fixed as of Babel 5.4.6, thanks!

**Comment by kevinb7** [comment]
@spicyj That makes sense.  I erroneously thought that the guard was why it was getting called a second time. 
@sebmck Thanks for the quick turn around!
